test/gtest/uct/ib/test_ib_event.cc

```diff
@@ -57,11 +57,16 @@ class uct_test_event_base : public uct_p2p_test {
     struct event_ctx {
         uct_ib_async_event_wait_t super;
         volatile bool             got;
+        uct_ib_device_t           *dev;
     };
 
     static unsigned last_wqe_check_cb(void *arg) {
         event_ctx *event = (event_ctx *)arg;
-        event->got       = true;
+
+        ucs_spin_lock(&event->dev->async_event_lock);
+        ucs_spin_unlock(&event->dev->async_event_lock);
```

Comment:
why need ucs_callbackq_remove_safe, since test already calls uct_ib_device_async_event_unregister

Reply:
We call uct_ib_device_async_event_unregister() only in uct_qp_test_event.
In uct_p2p_test_event ep init/cleanup does.
The problem in uct_p2p_test_event is that both the test and ep cleanup scheduling wait for event, so ep cleanup will clean only after itself.
Thus the test should cleanup after itself.

Reply:
since uct_ib_device_async_event_dispatch is uing async_event_lock to protect `entry->wait_ctx->cb_id = ucs_callbackq_add_safe` and uct_ib_device_async_event_unregister is using same lock to protect `ucs_callbackq_remove_safe(entry->wait_ctx->cbq, entry->wait_ctx->cb_id);`, I think the test should also put ucs_callbackq_remove_safe and `event->super.cb_id = UCS_CALLBACKQ_ID_NULL;` inside the lock.

we can say the hash table lock also protects all events which are in it.